webpack/assets/javascripts/react_app/components/componentRegistry.js

```diff
@@ -4,9 +4,92 @@ import StatisticsChartsList from './statistics/StatisticsChartsList';
 import PowerStatus from './hosts/powerStatus/';
 import NotificationContainer from './notifications/';
 import ToastsList from './toastNotifications/';
+import RelativeDateTime from './common/Date/RelativeDateTime';
+import LongDateTime from './common/Date/LongDateTime';
+import ShortDateTime from './common/Date/ShortDateTime';
+import Date from './common/Date/Date';
 import StorageContainer from './hosts/storage/vmware/';
 import forEach from 'lodash/forEach';
 import map from 'lodash/map';
+import { withIntlProvider } from '../common/i18n';
+import { Provider } from 'react-redux';
+
+function getDisplayName(Component) {
+  return Component.displayName || Component.name || 'Component';
+}
+
+function intlProviderWrapper() {
+  return withIntlProvider;
+}
+
+function storeProviderWrapper(store) {
+  return function (WrappedComponent) {
+    class StoreProvider extends React.Component {
+      render() {
+        return (
+          <Provider store={store}>
+            <WrappedComponent {...this.props} />
+          </Provider>
+        );
+      }
+    }
+    StoreProvider.displayName = `StoreProvider(${getDisplayName(WrappedComponent)})`;
+
+    return StoreProvider;
+  };
+}
+
+function dataProviderWrapper(data) {
+  return function (WrappedComponent) {
+    class DataProvider extends React.Component {
+      render() {
+        return (
+          <WrappedComponent data={data} />
+        );
+      }
+    }
+    DataProvider.displayName = `DataProvider(${getDisplayName(WrappedComponent)})`;
+
+    return DataProvider;
+  };
+}
+
+function propDataMapperWrapper() {
+  return function (WrappedComponent) {
+    class PropDataMapper extends React.Component {
+      render() {
+        return (
+          <WrappedComponent data={this.props} />
+        );
+      }
+    }
+    PropDataMapper.displayName = `PropDataMapper(${getDisplayName(WrappedComponent)})`;
+
+    return PropDataMapper;
+  };
+}
+
+class WrapperFactory {
```

Comment:
I believe that the WrapperFactory together with the functions above should be separate into a different file